JSDemos/Demos/DataGrid/UsingHeaderFilter/description.md

```diff
@@ -1,13 +1,13 @@
-A header filter allows users to select values from a predefined set and filter rows by these values. The values are listed in a pop-up menu. To open this menu, users can click header filter icons in column headers.
+A header filter allows users to filter rows by values selected in a pop-up menu. To display this menu, users can click header filter icons in column headers.
 
 ### Display Header Filter Icons
 
-Enable the **headerFilter**.[visible](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/#visible) property to display header filter icons for all columns. If you want to hide the icon for a specific column, disable the column's [allowHeaderFiltering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#allowHeaderFiltering) property.
+Assign **true** to the **headerFilter**.[visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/#visible) property to display header filter icons for all columns. To hide the icon for a specific column, set the column's [allowHeaderFiltering](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#allowHeaderFiltering) property to **false**.
 
 ### Enable Search Within Header Filters
 
-Users can search values within header filters. To display the search bar, assign **true** to the [allowSearch](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/#allowSearch) property in the global [headerFilter](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/headerFilter/) object or in a column's [headerFilter](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/headerFilter/) object. At the column level, you can also specify a comparison operation used to search header filter values ([searchMode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/headerFilter/#searchMode)).
+Users can search values within a header filter. To configure the search panel, define the [search](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/search/) property in the global [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) object or in a **columns**.[headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/) object. 
 
-### Change Available Filter Values
+You can compare the search string against specific fields in your data objects. For this purpose, apply the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. In this example, the search string is compared against the "City" and "State" fields.
```

Comment:
I don't understand how this works. If I'm searching in a header filter, then that filter is already populated with values from some field. Why would I want to specify a different field for search? 

Let's say I clicked the filter icon in the "Country" column. The values in the header filter are Brazil, Spain, Portugal. I also enabled the search panel. Now I set `search.searchExpr` to compare my search string against the City field. Not sure how all of this makes sense. Now I can type in Rio or Madrid and find something?

Reply:
Ok, let's try to figure out. For example, you have the following data: **[{ city: 'London', country: 'United Kingdom' }, { city: 'Berlin', country: 'German' }]**. If you specify the columns[].headerFilter.**searchExpr: ['City', 'Country']** expression, in the header filter's search bar you can enter 'London' or 'United Kingdom' to get the 'London' result. So, if you enter just country name, you get all cities of this country.

Reply:
@Abadzhev Please, take a look at my answer)

Reply:
What's the default behavior? Does the search include all fields, regardless of the column where I invoked the header filter?

Does the search panel even belong to a certain header filter? Maybe I just need to see a screenshot or demo.  

Here's what bothers me. Let's say I'm an end user. If I want to display all cities in a specific country. I would just search for a country. I would do that in the "Country" column. Entering a country name into the City column seems very weird to me. Why would anyone set up the control like that? :-)

The only way how all of this might make sense, is if the grid displayed just one single search panel... But you do mention that a search panel belongs to a header filter (a specific column). 

If you have a link to a demo I can see - it would be great.

Reply:
You can check the demo here: http://az-jsserver.corp.devexpress.com:81/Demos-23.1/WidgetsGallery/Demo/DataGrid/UsingHeaderFilter/

Reply:
How about this version? 

```suggestion
The search panel's default behavior is to look for values only within the same data field. You can include additional fields into the search. Specify the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. For example, this demo allows you to enter a state name into the City column's header filter. You can then see a list of all cities within the specified state, and then select city names that you want to use as a filter. 
```